tests/page-objects/messages/messages.po.js

```diff
@@ -1,25 +1,110 @@
 const helper = require('../../helper');
+const utils = require('../../utils');
+const common = require('../common/common.po');
 
 const sendMessageButton = element(by.className('mm-icon mm-icon-inverse mm-icon-caption send-message'));
 const exportButton = element(by.css('[ng-click=actionBar.left.exportFn()]'));
+const selectOptions = element.all(by.css('.select2-results__option'));
+const selectChoices = element.all(by.css('li.select2-selection__choice'));
+const sendAdditionalMessage = element(by.css('#message-footer .message-actions .btn-primary'));
+const messageTextArea = element(by.css('#message-footer textarea'));
+const addRecipient = element(by.css('.message-actions .btn.btn-link'));
+const selectedOption = element(by.css('#send-message select>option'));
+
+const openMessageContent = async (id, name) => {
+  await common.goToMessagesNative();
+  await helper.waitUntilReadyNative(module.exports.messageInList(id));
+  await helper.waitElementToPresentNative(module.exports.messageInList(id), 2000);
+  await clickLhsEntry(id, name);
+};
+
+const enterMessageText = async message => {
+  await element(by.css('#message-footer textarea')).click();
+  await helper.waitElementToBeVisible(sendAdditionalMessage);
+  await browser.wait(messageTextArea.sendKeys(message));
+};
+
+const searchSelect2 = async (searchText, totalExpectedResults, entrySelector, entryText) => {
+  await module.exports.messageRecipientSelect().sendKeys(searchText);
+  await browser.wait(async () => await selectOptions.count() === totalExpectedResults);
```

Comment:
Interested in what time you think is right? I agree 30 seconds is long but to me it doesn't make since to 1 off the timeout. Might make more sense to set the `defaultTimeoutInterval` to less. That would make all timeouts shorter though.

Reply:
I think waiting for most things longer than 5 seconds (like a button to appear or something uncomplicated like that) is over the top. When there are things that are known to take more than 5 seconds, we could add exceptions.

Reply:
So based on that, setting the default to 10 seconds and then the exception case would be when you know you need to wait for something long.

Reply:
I added the short timeout to 10 and it caused a bunch of failures. I think for now I'm going to leave that change out.

Reply:
Can you, please, add a timeout here?